src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -488,6 +488,22 @@ Copyright (C) 2011-2012 Xamarin. All rights reserved.
   </ItemGroup>
 </Target>
 
+<Target Name="_CheckAndroidHttpClientHandlerType"
+    Condition=" '$(AndroidHttpClientHandlerType)' != ''">
+  <PropertyGroup>
+    <ValidAndroidHttpClientHandlerTypes Condition="'$(UsingAndroidNETSdk)' != 'true'">Xamarin.Android.Net.AndroidClientHandler</ValidAndroidHttpClientHandlerTypes>
+    <ValidAndroidHttpClientHandlerTypes Condition="'$(UsingAndroidNETSdk)' != 'true'">$(ValidAndroidHttpClientHandlerTypes);System.Net.Http.MonoWebRequestHandler, System.Net.Http</ValidAndroidHttpClientHandlerTypes>
+    <ValidAndroidHttpClientHandlerTypes Condition="'$(UsingAndroidNETSdk)' != 'true'">$(ValidAndroidHttpClientHandlerTypes);System.Net.Http.HttpClientHandler, System.Net.Http</ValidAndroidHttpClientHandlerTypes>
+    <ValidAndroidHttpClientHandlerTypes Condition="'$(UsingAndroidNETSdk)' == 'true'">Xamarin.Android.Net.AndroidMessageHandler</ValidAndroidHttpClientHandlerTypes>
+    <ValidAndroidHttpClientHandlerTypes Condition="'$(UsingAndroidNETSdk)' == 'true'">$(ValidAndroidHttpClientHandlerTypes);System.Net.Http.SocketsHttpHandler, System.Net.Http</ValidAndroidHttpClientHandlerTypes>
+  </PropertyGroup>
+  <AndroidError Code="XA1031"
+      ResourceName="XA1031"
+      FormatArguments="$(AndroidHttpClientHandlerType)"
+      Condition=" !($(ValidAndroidHttpClientHandlerTypes.Contains ('$(AndroidHttpClientHandlerType)')))"
```

Comment:
Unfortunately, this check is too strict.  The [`$(AndroidHttpClientHandlerType)` MSBuild property](https://github.com/xamarin/xamarin-android/blob/main/Documentation/guides/building-apps/build-properties.md#androidhttpclienthandlertype) *allows* for arbitrary types:

> Controls the default `System.Net.Http.HttpMessageHandler` implementation which will be used by the `System.Net.Http.HttpClient` default constructor. The value is an assembly-qualified type name of an `HttpMessageHandler` subclass, suitable for use with [System.Type.GetType(string)](https://github.com/xamarin/xamarin-android/blob/main/dotnet/api/system.type.gettype#System_Type_GetType_System_String_)

The *intent* is that if someone *wishes* to provide their own `System.Net.Http.HttpMessageHandler` subclass, *they can do so* by:

 1. Setting `$(AndroidHttpClientHandlerType)` to `Their.HttpMessageHandler.Subclass, Assembly-Name`, and
 2. Setting `$(UseNativeHttpHandler)`=True.

*This* implementation does not allow that; it instead *requires* that *only* `Xamarin.Android.Net.AndroidMessageHandler` *or* `System.Net.Http.SocketsHttpHandler, System.Net.Http` be specified.

What we *actually* need to do is add a new Task/update an existing Task which is able to walk the inheritance chain of the type specified by `$(AndroidHttpClientHandlerType)` and verify that it fulfills our runtime requirements: https://github.com/xamarin/xamarin-android/blob/91669a4e489104668fdd07382e40cf070aa559ac/src/Mono.Android/Android.Runtime/AndroidEnvironment.cs#L356-L372